keras_cv/models/object_detection/retina_net/retina_net_inference_test.py

```diff
@@ -0,0 +1,190 @@
+        new_retina_net.load_weights(f"{tmp}/checkpoint.h5")
+
+        # manually check layers to make sure nothing is missed
+        for layer_original, layer_new in zip(
+            _get_retina_net_layers(retina_net), _get_retina_net_layers(new_retina_net)
+        ):
+            for weight, weight_new in zip(
+                layer_original.get_weights(), layer_new.get_weights()
+            ):
+                self.assertAllEqual(weight, weight_new)
+
+        # manually check layers to make sure nothing is missed in `get_weights()`
+        for layer_original, layer_new in zip(retina_net.layers, new_retina_net.layers):
+            for weight, weight_new in zip(
+                layer_original.get_weights(), layer_new.get_weights()
+            ):
+                self.assertAllEqual(weight, weight_new)
+
+    @pytest.mark.skipif(
+        "INTEGRATION" not in os.environ or os.environ["INTEGRATION"] != "true",
+        reason="Takes a long time to run, only runs when INTEGRATION "
+        "environment variable is set.  To run the test please run: \n"
+        "`INTEGRATION=true pytest keras_cv/",
+    )
+    def test_weight_loading_via_metrics(self):
+        x, y = _create_bounding_box_dataset(bounding_box_format="xywh")
+        retina_net, new_retina_net = _create_retina_nets(x, y, epochs=30)
+
+        tmp = tempfile.mkdtemp()
+        retina_net.save_weights(f"{tmp}/checkpoint.h5")
+        new_retina_net.load_weights(f"{tmp}/checkpoint.h5")
+
+        metrics = retina_net.evaluate(x, y, return_dict=True)
+        new_metrics = new_retina_net.evaluate(x, y, return_dict=True)
+
+        for key in metrics:
+            self.assertEqual(metrics[key], new_metrics[key])
+
+
+def _get_retina_net_layers(model):
+    return [
+        model.backbone,
+        model.feature_pyramid,
+        model.prediction_decoder,
+        model.anchor_generator,
+        model.label_encoder,
+        model.classification_head,
+        model.box_head,
+    ]
+
+
+def _create_retina_nets(x, y, epochs=1):
+    retina_net = keras_cv.models.RetinaNet(
+        classes=20,
+        bounding_box_format="xywh",
+        backbone="resnet50",
+        backbone_weights="imagenet",
+        include_rescaling=True,
+    )
+    retina_net.compile(
+        classification_loss=keras_cv.losses.FocalLoss(
+            from_logits=True,
+            reduction="none",
+        ),
+        box_loss=keras_cv.losses.SmoothL1Loss(l1_cutoff=1.0, reduction="none"),
+        optimizer="adam",
+        metrics=[
+            keras_cv.metrics.COCOMeanAveragePrecision(
+                class_ids=range(20),
+                bounding_box_format="xyxy",
+                name="MaP",
+            ),
+            keras_cv.metrics.COCORecall(
+                class_ids=range(20),
+                bounding_box_format="xyxy",
+                name="Recall",
+            ),
+        ],
+    )
+    retina_net.build((None, None, None, 3))
+    if epochs != 0:
```

Comment:
Might need some comment about why we need to train this model here (and not the model down below.)